src/tools/clippy/clippy_utils/src/hir_utils.rs

```diff
@@ -690,6 +693,9 @@ impl<'a, 'tcx> SpanlessHash<'a, 'tcx> {
                     }
                 }
             },
+            ExprKind::Let(_, ref expr, _) => {
```

Comment:
Why not also hash the pattern?

Reply:
Couldn't find an example that hashes patterns? Can you point one?

Reply:
No, but I think we should also hash `Pat`s, since we also implement `Eq` for it. This is the first `ExprKind` that has a `Pat` field. Could you add a `// FIXME` here, please? No need to implement `Pat` hashing in this PR.

Reply:
Ok. Do you mind also opening a PR on `rust-lang/clippy`? FIXMEs or TODOs tend to be forgotten over time.